packages/flutter/lib/src/rendering/table.dart

```diff
@@ -586,124 +558,47 @@ class RenderTable extends RenderBox {
       child.parentData = TableCellParentData();
   }
 
-  /// Replaces the children of this table with the given cells.
+  /// Insert child into this render object's child list after the given child.
   ///
-  /// The cells are divided into the specified number of columns before
-  /// replacing the existing children.
-  ///
-  /// If the new cells contain any existing children of the table, those
-  /// children are simply moved to their new location in the table rather than
-  /// removed from the table and re-added.
-  void setFlatChildren(int columns, List<RenderBox?> cells) {
```

Comment:
there's a chance that this and some of the other API changes will break someone. I would feel much more comfortable if the API was backwards compatible.

Reply:
the older API (setFlatChildren, addRow, setChild at x,y) is also potentially much more efficient than the new API